src/coreclr/jit/morph.cpp

```diff
@@ -13912,6 +13912,29 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
     }
 #endif
 
+#if defined(FEATURE_SIMD)
+    if (opts.OptimizationEnabled() && multiOp->OperIs(GT_SIMD))
+    {
+        bool allArgsAreConst = true;
+        for (GenTree* arg : multiOp->Operands())
+        {
+            if (!arg->OperIsConst())
```

Comment:
btw, I assume OperIsConst is also true for GenTreeVecCon? in that case constant vectors as arguments also won't be CSEd? (or GenTreeVecCon only for GT_HWINTRINSIC?)

Reply:
I guess this was under the assumption that `Vector.Create` methods do not have any vector parameter types?

Reply:
Yes, but your code doesn't check that it's Vector.Create or I am not reading it correctly?

Reply:
I see. I assumed `GT_SIMD` was only creation of vectors - that's not the case. I need to check for the constructors.

Reply:
Yes, but not for `Create`.

The check above of
> if (opts.OptimizationEnabled() && multiOp->IsVectorCreate())

should ensure that this is one of the Create calls that only takes primitive operands.